Sanguine Soul/sanguinesoul.asl

```diff
@@ -13,39 +13,42 @@ startup
 {
 	Assembly.Load(File.ReadAllBytes("Components/asl-help")).CreateInstance("Basic");
 	vars.Helper.GameName = "Sanguine Soul";
+	vars.Helper.Settings.CreateFromXml("Components/SanguineSoul.Settings.xml");
+
+	vars.CompletedSplits = new HashSet<string>();
+
 	vars.Helper.AlertRealTime();
-    vars.Helper.Settings.CreateFromXml("Components/SanguineSoul.Settings.xml");
 }
 
 init
 {
     // `fname` is the index into the GNames array
-    vars.CachedFNames = new Dictionary<long, string>();
-    vars.ReadFName = (Func<long, string>)(fname => 
+    var cachedFNames = new Dictionary<int, string>();
+    vars.ReadFName = (Func<int, string>)(fname => 
     {
-        if (vars.CachedFNames.ContainsKey(fname)) return vars.CachedFNames[fname];
+	string name;
+        if (cachedFNames.TryGetValue(fname, out name)) return name;
```

Comment:
Why not just use .ContainsKey?

Reply:
We don't want to index the dictionary more than once.

Reply:
Why not? Is there a substantive performance improvement in doing so?

Reply:
It is technically faster, but in my opinion it reads worse. I don't see a case for changing it.

If we were about performance improvements, I don't think this would be a good place to start.

Reply:
In addition to be more performant, it's simply also better practice. The current code also may have a TOC/TOU issue (which may not seem important to you, but an issue like that potentially existing at all should be reason to fix it).

Reply:
That's fair.